File: Programmers/Level 1/크레인 인형뽑기 게임/Solution.py
def solution(board, moves):
    answer = 0
    basket = []
    
    ## 인형 쌓기
    for m in moves:
        for b in board:
            if b[m-1] != 0:
                basket.append(b[m-1])
                b[m-1] = 0
                break
    
    ## 인형 터트리기
    i = 0
    while len(basket)>1 and len(basket)-1 != i:
        if basket[i] == basket[i+1]:
            answer += 2
            basket[i:i+2] = []
            i -= 1
        else:
            i += 1

    return answer

File: Programmers/Level 1/크레인 인형뽑기 게임/Solution.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<vector<int>> board, vector<int> moves) {
    int answer = 0;
    vector<int> results(1);
    
    for(int i=0;i<moves.size();i++){
        for(int j=0;j<board.size();j++){
            if(board[j][moves[i]-1]!=0){
                if(board[j][moves[i]-1]==results.back()){
                    answer+=2;
                    results.pop_back();
                }
                else
                    results.push_back(board[j][moves[i]-1]);
                board[j][moves[i]-1]=0;
                break;
            }
        }
    }
    
    return answer;
}